### PR TITLE
PRS-ACtionMailer&HelperMethod&ActiveJob  set the Action Mailer and th…

### PR DESCRIPTION
**Pull request by louy755**
…e helper Methos also add the active Job setting

### CODE REVIEW

File: lib/ecrs_helpers.rb [hunk 1]
```diff
@@ -134,56 +134,56 @@ def get_data(browser, contractor_id, start_date, end_date, counter, file_path, m
...
+              HelperMethods::Logger.app_logger('info', "file deleted - #{del_res}")
               return true, number_of_records, nil
             end
 
           else
             return true, 0, nil
           end
         rescue Exception => e
-          Rails.logger.warn e.message
+           HelperMethods::Logger.app_logger('warn', e)
```
Comment (ssharba) [inline lib/ecrs_helpers.rb]: this is error not warn.

File: lib/ecrs_helpers.rb [hunk 2]
```diff
@@ -134,56 +134,56 @@ def get_data(browser, contractor_id, start_date, end_date, counter, file_path, m
...
             if sys_msg
               File.delete(file)
               sys_msg = system "unix2dos #{result_file}"
-              Rails.logger.info "unix2dos (#{result_file}) - #{sys_msg}"
+              HelperMethods::Logger.app_logger('info', "unix2dos (#{result_file}) - #{sys_msg}")
             end
           end
         rescue Exception => e
-          Rails.logger.warn e.message
+           HelperMethods::Logger.app_logger('warn', e)
```
Comment (ssharba) [inline lib/ecrs_helpers.rb]: error not warn.

File: lib/ecrs_helpers.rb [hunk 3]
```diff
@@ -120,10 +120,10 @@ def get_contractor_data(browser, contractor_id, client_id, ecrs_task_id, file_pa
               number_of_days_per_period = 30
             end
           end
-          Rails.logger.info "returning contractor info: #{files_array}"
+          HelperMethods::Logger.app_logger('info', "returning contractor info: #{files_array}")
           return files_array
         rescue Exception => e
-          Rails.logger.warn e.message
+           HelperMethods::Logger.app_logger('warn', e)
```
Comment (ssharba) [inline lib/ecrs_helpers.rb]: error not warn

File: lib/tasks/ecrs_tasks.rake [hunk 4]
```diff
@@ -21,47 +21,47 @@ namespace :ecrs_tasks do
...
-            Rails.logger.info "Calling file merger"
+            HelperMethods::Logger.app_logger('info', "File: #{files_array}")
+            HelperMethods::Logger.app_logger('info', "Calling file merger")
             EcrsHelpers::Contractors.merge_files(files_array, c.contractor_id, file_path, today) unless files_array.empty?
-            Rails.logger.info "Done mergeing files"
+            HelperMethods::Logger.app_logger('info', "Done mergeing files")
             browser.goto ENV['ECRS_NEW_CONTRACTOR_URL']
           rescue Exception => e
-            Rails.logger.warn e.message
+             HelperMethods::Logger.app_logger('warn', e)
```
Comment (ssharba) [inline lib/tasks/ecrs_tasks.rake]: error

File: lib/tasks/ecrs_tasks.rake [hunk 5]
```diff
@@ -21,47 +21,47 @@ namespace :ecrs_tasks do
...
             browser.goto ENV['ECRS_NEW_CONTRACTOR_URL']
           end
         end
       end
-      Rails.logger.info "summary was sent to #{ENV['SUMMARY_RECEIVERS']}"
+      HelperMethods::Logger.app_logger('info', "summary was sent to #{ENV['SUMMARY_RECEIVERS']}")
       # EcrsHelpers::Email.send_email(ENV['ECRS_EMAIL_ADDRESS'], ENV['ECRS_EMAIL_PASSWORD'], ENV['SUMMARY_RECEIVERS'], "[#{ENV['ENVIRONMENT']}] ECRS Summary report", summary)
     rescue Exception => e
-      Rails.logger.warn e.message
+      HelperMethods::Logger.app_logger('warn', e)
```
Comment (ssharba) [inline lib/tasks/ecrs_tasks.rake]: error

File: lib/ecrs_helpers.rb (hunk 3 above)
Comment (louy755) [inline lib/ecrs_helpers.rb]: will commit the changes now

File: lib/ecrs_helpers.rb (hunk 1 above)
Comment (louy755) [inline lib/ecrs_helpers.rb]: ok

File: lib/helper_method.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+module HelperMethods
+
+	class Logger
+	  class << self
+	  def app_logger(type, e)
```
Comment (ssharba) [inline lib/helper_method.rb]: instead of e, I would call it message here.

File: lib/helper_method.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+	    case type
+	     when type = 'error'
+	     	if Rails.env.production?
+	            NotificationMailer.app_error(e.message).deliver_later
+	        end
+	            Rails.logger.debug "==> error send_fax: #{e} <=="
```
Comment (ssharba) [inline lib/helper_method.rb]: "==> error send_fax: #{e} <=="
this will be wrong if it was "info" type.

**Pull request merged**